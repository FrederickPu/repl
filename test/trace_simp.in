{"cmd": "def f := 37"}

{"cmd": "@[simp] theorem f_def : f = 37 := by rfl"}

{"cmd": "set_option trace.Meta.Tactic.simp true"}

{"cmd": "example : f = 37 := by simp"}

{"cmd": "example : f = 37 := by sorry", "env": 2}

{"tactic": "trace \"37\"", "proofState": 0}

{"tactic": "simp", "proofState": 0}

{"cmd": "example : True := by\n  trace \"!\"\n  trivial"}
