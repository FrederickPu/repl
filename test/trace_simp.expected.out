{"results": [{"source": "def f :=\n  37", "env": 0}]}

{"results": [{"source": "@[simp]\ntheorem f_def : f = 37 := by rfl", "env": 1}]}

{"results": [{"source": "set_option trace.Meta.Tactic.simp true", "env": 2}]}

{"results":
 [{"source": "example : f = 37 := by simp",
   "messages":
   [{"severity": "info",
     "pos": {"line": 1, "column": 23},
     "endPos": {"line": 1, "column": 27},
     "data":
     "[Meta.Tactic.simp.rewrite] f_def:1000, f ==> 37\n[Meta.Tactic.simp.rewrite] @eq_self:1000, 37 = 37 ==> True"}],
   "env": 3}]}

{"results":
 [{"source": "example : f = 37 := by sorry",
   "sorries":
   [{"proofState": 0,
     "pos": {"line": 1, "column": 23},
     "goalsBefore": "⊢ f = 37",
     "endPos": {"line": 1, "column": 28}}],
   "messages":
   [{"severity": "warning",
     "pos": {"line": 1, "column": 0},
     "endPos": {"line": 1, "column": 7},
     "data": "declaration uses 'sorry'"}],
   "env": 4}]}

{"traces": ["37"], "proofState": 1, "goalsAfter": ["⊢ f = 37"]}

{"traces":
 ["[Meta.Tactic.simp.rewrite] f_def:1000, f ==> 37",
  "[Meta.Tactic.simp.rewrite] @eq_self:1000, 37 = 37 ==> True"],
 "proofState": 2,
 "goalsAfter": []}

{"results":
 [{"source": "example : True := by\n  trace \"!\"\n  trivial",
   "messages":
   [{"severity": "info",
     "pos": {"line": 2, "column": 2},
     "endPos": {"line": 2, "column": 7},
     "data": "!"}],
   "env": 5}]}

